src/rno-g-acq.c

```diff
@@ -583,6 +583,15 @@ static int did_bias_scan = 0;
 
 static int do_bias_scan() 
 {
+  printf("Performing timing measurements. This should just take a few minutes (but bias scan is coming next ...).");
+
+  string filename = "/home/rno-g/stationrc/record_timings.py ";
+  string command = "python ";
+  string flags = "-n 5 --data_dir /power/timing";
+
+  command += filename + flags;
+  system(command.c_str());
```

Comment:
So this is is probably very dangerous, because radiant_open has already been called, and there's no guarantee that the RADIANT is left in state that is equivalent. In particular, the SPI mode the python code would use is quite different, though perhaps this script doesn't use SPI readout. It's possible that the UART is opened in a compatible way but I don't know for sure. 

Also, a number of RADIANT settings are set between radiant_open and this call (e.g. in radiant_initial_setup,  where the LABS are stopped, the counters are reset and radiant_set_internal_triggers_per_cycle is called, leading to potentially  wrong behavior.

In general, I'd favor that the python code implement the same advisory locks that the C code does (using flock) and any calling of python radiant things be prior to radiant_open, otherwise I think it's very hard to make sure nothing goes wrong (and even if it works now, that it keeps working in the future...)

Reply:
have a look!